Rebase 3.32 (code)

uajain opened the pull request:
Rebase list: (Updated again on 16-April-2019)
* pick 51e06357 Update Afrikaans translation
**drop** as translation commit changed in upstream 
* pick 76d64a8b Debian-specific patch: add /var/lib/doc-base to the list of dirs that librarian will parse.  
**drop** as debian specific patch from `patch/` directory
* pick 5266499b Added Endless customizations (CSS, images, fonts) 
**keep**
* pick 7a38488a Added script to validate links in our help documentation 
**keep**  But also address it as a part of https://phabricator.endlessm.com/T25386 (read comment below)
* pick 69486905 Link with libm 
~~**keep**~~ Drop
* pick 1da6b979 Fix wrong generator rule for target data/domains/yelp.pot 
**keep** but Squash with "Added Endless customizations"
* pick 79344fe4 Add Transifex translation infrastructure 
**drop**
* pick 5e3870ab Commit .pot file 
**drop**
* pick 1b5ab9b6 Remove Video Library 
**keep** but Squash with "Added Endless customizations"
* pick dce38086 window: Always show the search bar 
**keep**
* pick 478c5b64 project: Replace tutorial by feedback 
**keep** but squash with "Added Endless customizations"


========== Below commits are dropped in favour of [search-provider upstream MR](https://gitlab.gnome.org/GNOME/yelp/merge_requests/2) ===========
* pick c67bb897 search-provider: add search provider 
* pick 93c3b904 libyelp: add env variable YELP_SKIP_GUI 
* pick 95177b98 *: make application D-Bus launchable 
* pick 1bf9ec3e libyelp: keep parent URI alive 
* pick bbdf4e25 *: export default uri as YELP_GNOME_HELP_URI 
* pick b4260089 src: handle --gapplication-service flag 
* pick b31e251d src: add actions to yelp 
* pick 6ab16707 gitignore: update file 
* pick a4379142 search_provider: use g_clear_object instead 
* pick 2628b58b search-provider: address icon related comments 
* pick 612bb3d4 libyelp: fix missing space 
* pick c6c3c8c0 search_provider: unref app in main 
* pick 4806759e search-provider: use g_dbus_method_invocation_take_error 
* pick a4508183 search-provider: remove unnecessary casting 
* pick de0f06df search-provider: remove unnecessary initialization 
* pick b2ee8b47 search-provider: free page_id 
* pick 5001a301 search-provider: make env_var a const 
* pick 4c7f43e5 search-provider: unref base_uri 
* pick fd103334 search-provider: add to EXTRA_DIST 
* pick 99a92c8b search-provider: add CLEANFILES 
* pick 12f8154c Makefile: add org.gnome.Yelp.service to CLEANFILES 
* pick bf1b049d search-provider: pass inactivity timeout directly 
* pick 593e6b36 libyelp: add unref uri 
* pick 23f20fcb search-provider: directly call when map is populated 
* pick b2a7ceba search-provider: pass pointer to clear 
* pick 66a22508 search-provider: cast GDestroyNotify 
* pick 33cffa5b search-provider: switch to g_signal_connect_object
* pick 31acaf17 search-provider: drop app and don't cast it to pointer
* pick a4e6d9ad search-provider: remove NULL case for meta results
* pick f467c1da search-provider: remove unnecessary file from CLEAN_FILES
* pick 9c2ac242 search-provider: merge search conditions
* pick 1e8a69ae search-provider: free delayed_result_getters in dispose
* pick 646ced66 search-provider: remove all elements and not destroy
* pick 37deaae7 search-provider: add release boolean
* pick d7c19162 src: handle gapplication flag
* pick e3667c4b *: remove env var workaround
* pick 2145497b search-provider: style
* pick 96e6921b search-provider: enclose condition with braces
* pick 99089d36 search-provider: do not pass address of delayed
* pick ccebb3eb search-provider: extract function for initial and subsearch
* pick b1bb5712 search-provider: coding style fix
* pick 4592b8fe search-provider: use correct include paths
* pick 32e6c46e search-provider: replace signals
* pick c60f446b desktop: remove DBusActivatable
* pick 6087b631 Desktop file: temporary hack for downstream name and icon
* pick c98b9db4 search-provider: Serialize an instance of GFileIcon, not a GThemedIcon
* pick 9ff756f1 application: fix crash when activating GActions
* pick 755a90cc Revert "desktop: remove DBusActivatable"
* pick 47e797e9 application: make sure to create a window in activate()
* pick ed8a246a build: Fix distcheck
=========================================================================

* pick 2a213e3f Update .pot file
**drop**
* pick 1fc95104 Add translation support for several new languages
**drop**
* pick 34d48524 Latest translations from Transifex
**drop**
* pick b15ef5a8 Merged with downstream translations in Transifex
**drop**
* pick 3e53cd3a Updates from Transifex
**drop**
* pick bb41c143 yelp-uri: Parse extra slashes after colon in resolve_help_uri()
**keep**
* pick b4c2530e yelp-application: Allow YelpApplication to handle opening files
**keep**
* pick 1e927f90 Fix translation of sidebar
**drop**. (as @wjt suggests below in the comments)
* pick f04c5a57 Pulled sidebar translations from Transifex
**drop**

Update:
New patches: 
1e2aa404 : From debian packaging in `patch/` directory.
8aebfe24 : `configure.ac` patched as discussed [here](https://phabricator.endlessm.com/T25371#693641)
59b2c298 : Picked memory-leaks-fix from upstream [commit](https://gitlab.gnome.org/GNOME/yelp/commit/5956af92d2c2cb7030127ef6fa8e744853528568)

https://phabricator.endlessm.com/T25371

uajain commented:
Diffs:

* [git diff -p --stat upstream/gnome-3-26..master -- ':!po' ':!.tx'](https://gist.github.com/uajain/30d773057c5ad6507cb47adc674f5b92)

* [git diff -p --stat upstream-master..uajain/T25371-rebase
](https://gist.github.com/uajain/4abad245fc191ce3690deaf939f2ab4e)

wjt commented:
pick 51e0635 Update Afrikaans translation
drop as translation not present in upstream

I don't really understand this note. The common ancestor with upstream is 059ede65; this commit is before that. I agree that it doesn't need to be picked because it is already part of the history.

pick 76d64a8 Debian-specific patch: add /var/lib/doc-base to the list of dirs that librarian will parse.
drop as debian specific patch from patch/ directory

OK – I think it's right that we don't need this.
    
pick 5266499 Added Endless customizations (CSS, images, fonts)
keep

How do these look?

pick 7a38488 Added script to validate links in our help documentation
keep

I think this script would be more useful if it was run as part of the build process for the documentation itself. However, let's do that as part of https://phabricator.endlessm.com/T25386 for 3.6.1 or so.

pick 6948690 Link with libm
keep

The commit message says:

> This shouldn't be needed. Why is this needed?

Did you investigate this? Is it needed?

pick 1da6b97 Fix wrong generator rule for target data/domains/yelp.pot
keep

If this change is correct, it should go upstream.

pick 79344fe Add Transifex translation infrastructure
drop
pick 5e3870a Commit .pot file
drop

OK

pick 1b5ab9b Remove Video Library
keep

This removes downstream code that was added in "Added Endless customizations (CSS, images, fonts)". It could be squashed into it.

pick dce3808 window: Always show the search bar
keep

OK

pick 478c5b6 project: Replace tutorial by feedback
keep

Again, this could perhaps be squashed into "Added Endless customizations".

pick c67bb89 search-provider: add search provider
keep

pick 93c3b90 libyelp: add env variable YELP_SKIP_GUI
keep

This is reverted by "*: remove env var workaround".

More generally, all these patches that deal with the search provider are very different to the versions on https://gitlab.gnome.org/GNOME/yelp/merge_requests/2 . Which is the newer version of the series? I think the version in that MR is likely to be the newer one, since it squashes away lots of the commits here which fix bugs introduced in previous downstream changes.

If the open MR incorporates all of the changes here, in a neater form, we should take those patches downstream. I'll skip over all the other search provider-related changes.

pick 95177b9 *: make application D-Bus launchable
keep
pick 1bf9ec3 libyelp: keep parent URI alive
keep
pick bbdf4e2 *: export default uri as YELP_GNOME_HELP_URI
keep
pick b426008 src: handle --gapplication-service flag
keep
pick b31e251 src: add actions to yelp
keep
pick 6ab1670 gitignore: update file
keep
pick a437914 search_provider: use g_clear_object instead
keep
pick 2628b58 search-provider: address icon related comments
keep
pick 612bb3d libyelp: fix missing space
keep
pick c6c3c8c search_provider: unref app in main
keep
pick 4806759 search-provider: use g_dbus_method_invocation_take_error
keep
pick a450818 search-provider: remove unnecessary casting
keep
pick de0f06d search-provider: remove unnecessary initialization
keep
pick b2ee8b4 search-provider: free page_id
keep
pick 5001a30 search-provider: make env_var a const
keep
pick 4c7f43e search-provider: unref base_uri
keep
pick fd10333 search-provider: add to EXTRA_DIST
keep
pick 99a92c8 search-provider: add CLEANFILES
keep
pick 12f8154 Makefile: add org.gnome.Yelp.service to CLEANFILES
keep
pick bf1b049 search-provider: pass inactivity timeout directly
keep
pick 593e6b3 libyelp: add unref uri
keep
pick 23f20fc search-provider: directly call when map is populated
keep
pick b2a7ceb search-provider: pass pointer to clear
keep
pick 66a2250 search-provider: cast GDestroyNotify
keep
pick 33cffa5 search-provider: switch to g_signal_connect_object
keep
pick 31acaf1 search-provider: drop app and don't cast it to pointer
keep
pick a4e6d9a search-provider: remove NULL case for meta results
keep
pick f467c1d search-provider: remove unnecessary file from CLEAN_FILES
keep
pick 9c2ac24 search-provider: merge search conditions
keep
pick 1e8a69a search-provider: free delayed_result_getters in dispose
keep
pick 646ced6 search-provider: remove all elements and not destroy
keep
pick 37deaae search-provider: add release boolean
keep
pick d7c1916 src: handle gapplication flag
keep
pick e3667c4 *: remove env var workaround
keep
pick 2145497 search-provider: style
keep
pick 96e6921 search-provider: enclose condition with braces
keep
pick 99089d3 search-provider: do not pass address of delayed
keep
pick ccebb3e search-provider: extract function for initial and subsearch
keep
pick b1bb571 search-provider: coding style fix
keep
pick 4592b8f search-provider: use correct include paths
keep
pick 32e6c46 search-provider: replace signals
keep



pick c60f446 desktop: remove DBusActivatable
drop as reverted in 755a90c

OK

pick 6087b63 Desktop file: temporary hack for downstream name and icon
keep

I don't like the idea of carrying forward a patch that was declared "temporary" in 2017.

pick c98b9db search-provider: Serialize an instance of GFileIcon, not a GThemedIcon
keep
pick 9ff756f application: fix crash when activating GActions
keep

More stuff that should be part of the upstream search provider MR.

pick 755a90c Revert "desktop: remove DBusActivatable"
drop

OK

pick 47e797e application: make sure to create a window in activate()
keep

More search provider stuff, I think.

pick ed8a246 build: Fix distcheck
keep

This fixes bugs introduced by previous patches in the series and should be squashed into them.

pick 2a213e3 Update .pot file
drop
pick 1fc9510 Add translation support for several new languages
drop
pick 34d4852 Latest translations from Transifex
drop
pick b15ef5a Merged with downstream translations in Transifex
drop
pick 3e53cd3 Updates from Transifex
drop


pick bb41c14 yelp-uri: Parse extra slashes after colon in resolve_help_uri()
keep

Based on the fact that this adds -Wno-error=comment, I think this patch must introduce compiler warnings. Let's fix those warnings, instead.

pick b4c2530 yelp-application: Allow YelpApplication to handle opening files
keep

pick 1e927f9 Fix translation of sidebar
drop

There are some useful instructions here that look like they will need to be followed by hand... not great. Please can we fix the dependencies in the makefiles instead?

> Fix translation of sidebar
> 
> There is a separate pot file in data/domains that has to be
> generated in addition to the one in the po directory.
> Then, the one in the po file needs to be generated afterwards.
> This can be accomplished via the following steps:
>   ./autogen.sh
>   rm data/domains/yelp.pot
>   make data/domains/yelp.pot
>   cd po
>   rm yelp.pot
>   make yelp.pot
> 
> https://phabricator.endlessm.com/T4336
> https://phabricator.endlessm.com/T22631


pick f04c5a5 Pulled sidebar translations from Transifex
drop


Update:
New patches:
0648476 : From debian packaging in patch/ directory.
  There's no patch with this commit ID in this series but I think this refers to:
  
  commit 08e4a3946e600112f0631a581f4bcda0733ba0d9 (HEAD -> uajain/T25371-rebase)
  Author: Jeremy Bicha <jbicha@ubuntu.com>
  Date:   Mon Sep 26 21:08:21 2011 -0400

  Remove "Search for packages" 404 feature
  
  which seems OK
  
5a806a8 : configure.ac patched as discussed here
1178217 : Picked memory-leaks-fix from this open PR (Merge requests for those already submitted upstream. Read the commit message)

This is several unrelated changes smushed together.

* The patch from https://gitlab.gnome.org/GNOME/yelp/merge_requests/13/diffs?commit_id=5956af92d2c2cb7030127ef6fa8e744853528568 which has been merged should be cherry-picked verbatim.
* If the search provider change is part of Cosimo's MR, then great. We should be cherry-picking that MR, not our old "X, Y, fix X, fix Y" patches.
No...

wjt commented:
> How do these look?

![Screenshot from 2019-04-12 17-11-08](https://user-images.githubusercontent.com/86760/56051284-008f5c80-5d46-11e9-827f-ae5135b675f2.png)

There's missing padding in each row. On 3.5 the horizontal padding between boxes matches the vertical padding.

![Screenshot from 2019-04-12 17-13-19](https://user-images.githubusercontent.com/86760/56051387-506e2380-5d46-11e9-85ab-f7420c2ded29.png)

The alignment here is wrong – the headers have underlines but the boxes are centered and flow past them.

uajain commented:
> pick 7a38488 Added script to validate links in our help documentation
keep
I think this script would be more useful if it was run as part of the build process for the documentation itself. However, let's do that as part of https://phabricator.endlessm.com/T25386 for 3.6.1 or so.

I have dropped this for now. Should I link this comment on T25386?

>pick 6948690 Link with libm
keep
The commit message says:
This shouldn't be needed. Why is this needed?
Did you investigate this? Is it needed

Yes should **not** be needed.

>pick 1da6b97 Fix wrong generator rule for target data/domains/yelp.pot
keep
If this change is correct, it should go upstream.

How? This seems purely a endless change/correction to me. I have squashed this with "Add Endless customization"

>pick 6087b63 Desktop file: temporary hack for downstream name and icon
keep
I don't like the idea of carrying forward a patch that was declared "temporary" in 2017.

Dropping this with a bit of uncertainty. I better take a note of it and test it on master-image after this passes code review. The changes in the `.desktop` should be addressed via eos-shell-content but I am not really sure how these got merged with other changes in this branch (like DBusActivatable=true).


I need help on this one:
>pick 1e927f9 Fix translation of sidebar
drop
There are some useful instructions here that look like they will need to be followed by hand... not great. Please can we fix the dependencies in the makefiles instead?
>Fix translation of sidebar
There is a separate pot file in data/domains that has to be
generated in addition to the one in the po directory.
Then, the one in the po file needs to be generated afterwards.
This can be accomplished via the following steps:
./autogen.sh
rm data/domains/yelp.pot
make data/domains/yelp.pot
cd po
rm yelp.pot
make yelp.pot
https://phabricator.endlessm.com/T4336
https://phabricator.endlessm.com/T22631

wjt commented:
> > pick [7a38488](https://github.com/endlessm/yelp/commit/7a38488a4ac0648d90b6bf57de9ea4b44f18d644) Added script to validate links in our help documentation
> > keep
> > I think this script would be more useful if it was run as part of the build process for the documentation itself. However, let's do that as part of https://phabricator.endlessm.com/T25386 for 3.6.1 or so.
> 
> I have dropped this for now. Should I link this comment on T25386?

I'd prefer that we keep this patch so we don't lose this script – I just meant that it would be even better if the documentation ran this check on itself. Please keep this patch, and also comment on that ticket.

> > pick [6948690](https://github.com/endlessm/yelp/commit/694869051c077eb08c887c25b845b2f96b275961) Link with libm
> > keep
> > The commit message says:
> > This shouldn't be needed. Why is this needed?
> > Did you investigate this? Is it needed
> 
> Yes should **not** be needed.

Hooray!

> > pick [1da6b97](https://github.com/endlessm/yelp/commit/1da6b979f8600b28318b627904f691a6d553750f) Fix wrong generator rule for target data/domains/yelp.pot
> > keep
> > If this change is correct, it should go upstream.
> 
> How? This seems purely a endless change/correction to me. I have squashed this with "Add Endless customization"

You're right: for some reason I missed the word "endless" in the diff! Squashing sounds good.

> 
> > pick [6087b63](https://github.com/endlessm/yelp/commit/6087b631217e1859509de052b605e663b86b21a4) Desktop file: temporary hack for downstream name and icon
> > keep
> > I don't like the idea of carrying forward a patch that was declared "temporary" in 2017.
> 
> Dropping this with a bit of uncertainty. I better take a note of it and test it on master-image after this passes code review. The changes in the `.desktop` should be addressed via eos-shell-content but I am not really sure how these got merged with other changes in this branch (like DBusActivatable=true).

Sounds good. The files are merged at package build time by https://github.com/endlessm/eos-shell-content/blob/master/tools/eos-content-merge via https://github.com/endlessm/eos-shell-content/blob/master/tools/dh_eoscontent#L154-L158 . If it turns out that this does not work, I'll investigate while doing the translations.

> > pick [1e927f9](https://github.com/endlessm/yelp/commit/1e927f90faae97ebb1b8823d2cfee2fb346e87cf) Fix translation of sidebar
> > drop
> > There are some useful instructions here that look like they will need to be followed by hand... not great. Please can we fix the dependencies in the makefiles instead?
> > Fix translation of sidebar
> > There is a separate pot file in data/domains that has to be
> > generated in addition to the one in the po directory.
> > Then, the one in the po file needs to be generated afterwards.
> > This can be accomplished via the following steps:
> > ./autogen.sh
> > rm data/domains/yelp.pot
> > make data/domains/yelp.pot
> > cd po
> > rm yelp.pot
> > make yelp.pot
> > https://phabricator.endlessm.com/T4336
> > https://phabricator.endlessm.com/T22631


I took a look, and I think probably the best use of time is to keep this as a manual process. Let's put a link to the old commit (with instructions) onto https://phabricator.endlessm.com/T25388 and drop this commit. I'll do it as part of the regular "re-add Transifex to all projects" flow.

uajain commented:
@wjt, thanks for reading comment. Turns out I am reaching a mental saturation with this rebase and still couldn't figure out where/how the CSS changes are to be done to fix the UI. :/ 
It's probably in `data/endless/css/article-style.css` but still it's a huge file to figure things out.

I think I need a break and in that case I am pushing everything that I have locally. Sounds good?

wjt commented:
Sounds good to me!

I found https://gitlab.gnome.org/GNOME/yelp/merge_requests/8 which would make it possible to launch the Web Inspector. Cherry-picking that (in your working copy) might make it a bit easier to track down the necessary CSS!

uajain commented:
> Sounds good to me!

Forced pushed my local rebase-branch. 

> I found https://gitlab.gnome.org/GNOME/yelp/merge_requests/8 which would make it possible to launch the Web Inspector. Cherry-picking that (in your working copy) might make it a bit easier to track down the necessary CSS!
Ah, great. Let me take a look later today / tomorrow.

wjt commented:
```
mv -f .deps/yelp_search_provider-yelp-shell-search-provider-generated.Tpo .deps/yelp_search_provider-yelp-shell-search-provider-generated.Po
../../../search-provider/yelp-search-provider.c:30:10: fatal error: libyelp/yelp-settings.h: No such file or directory
 #include "libyelp/yelp-settings.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~
```

perhaps this file is not listed in the source files in the makefile.am so is missing from the `make dist` tarball

uajain commented:
I fixed the remaining bits:
* Jenkins build issue: turns out it was a error in cosimoc's branch : Left a comment on [upstream MR](https://gitlab.gnome.org/GNOME/yelp/merge_requests/2#note_490911) and fixed in this PR.
* CSS issues. Padding and Margin issues @wjt pointed out in the screenshot. Squashed in "Add Endless customizations"
* Kept around `pick 7a38488 Added script to validate links in our help documentation` as @wjt mentioned in this [comment](https://github.com/endlessm/yelp/pull/53/#issuecomment-483634302).

I think these were all the outstanding blockers. If I have missed any, please bring it to my notice.
-> CR ready.

wjt commented on libyelp/yelp-uri.c:
```diff
@@ -846,20 +857,25 @@ resolve_ghelp_uri (YelpUri *uri)
 resolve_help_uri (YelpUri *uri)
 {
-    /* help:document[/page][?query][#frag]
+    /* help:[/*]document[/page][?query][#frag]
```
> pick [bb41c14](https://github.com/endlessm/yelp/commit/bb41c143338b08fe1a85846a0e5ba4f296b0d106) yelp-uri: Parse extra slashes after colon in resolve_help_uri()
> keep
> 
> Based on the fact that this adds -Wno-error=comment, I think this patch must introduce compiler warnings. Let's fix those warnings, instead.

This wasn't addressed. I've looked a little closer and I'd guess that the problem is on this line, where the string `/*` appears in a comment:
```
libyelp/yelp-uri.c: In function ‘resolve_help_uri’:
libyelp/yelp-uri.c:860:14: warning: "/*" within comment [-Wcomment]
     /* help:[/*]document[/page][?query][#frag]
```

I suggest changing this line to:

```suggestion
    /* help:[/+]document[/page][?query][#frag]
```

and dropping the `-Wno-error=comment` changes to the makefile.

uajain replied:
Addressed, squashed and pushed.
